cypress/integration/tx_modal.spec.js

```diff
@@ -0,0 +1,161 @@
+        cy.contains(RECIPIENT_ADDRESS).should('be.visible')
+      })
+
+      it('should have all tokens available in the token selector', () => {
+        // Click on the Token selector
+        cy.contains('Select an asset*').click()
+
+        const ownedTokens = ['Dai', 'Wrapped Ether', 'Ether', 'Uniswap', 'Gnosis', '0x', 'USD Coin']
+        ownedTokens.forEach((token) => {
+          cy.get('ul[role="listbox"]').contains(token)
+        })
+      })
+
+      it('should validate token amount', () => {
+        // Select a token
+        cy.get('ul[role="listbox"]').contains('Gnosis').click()
+
+        // Insert an incorrect amount
+        cy.get('input[placeholder="Amount*"]').type('0.4')
+
+        // Selecting more than the balance is not allowed
+        cy.contains('Maximum value is 0.000004')
+
+        // Form field contains an error class
+        cy.get('input[placeholder="Amount*"]')
+          // Parent div is MuiInputBase-root
+          .parent('div')
+          .should(($div) => {
+            // Turn the classList into an array
+            const classList = Array.from($div[0].classList)
+            expect(classList).to.include('MuiInputBase-root').and.to.include('Mui-error')
+          })
+
+        // Insert a correct amount
+        cy.get('input[placeholder="Amount*"]').clear().type('0.000002')
+
+        // Form field does not contain an error class
+        cy.get('input[placeholder="Amount*"]')
+          // Parent div is MuiInputBase-root
+          .parent('div')
+          .should(($div) => {
+            // Turn the classList into an array
+            const classList = Array.from($div[0].classList)
+            // Check if it contains the error class
+            expect(classList).to.include('MuiInputBase-root').and.not.to.include('Mui-error')
+          })
+
+        // Click Send max fills the input with token total amount
+        cy.contains('Send max').click()
+        cy.get('input[placeholder="Amount*"]').should('have.value', '0.000004')
+      })
+
+      it('should advance to the Review step', () => {
+        // Clicks Review
+        cy.contains('Review').click()
+
+        // Modal step 2
+        cy.contains('Step 2 of 2').should('be.visible')
+      })
+    })
+
+    describe('Review modal', () => {
+      before(() => {
+        // Wait max 10s for estimate to finish
+        cy.contains('Submit', { timeout: 10000 })
+      })
+
+      it('should have the same parameters as the previous step', () => {
+        // Sender
+        cy.contains('Sending from').parent().next().contains(TEST_SAFE)
+        // Recipient
+        cy.contains('Recipient').parent().next().contains(RECIPIENT_ADDRESS)
+
+        // Token value
+        cy.contains('0.000004 GNO')
+      })
+
+      it('should contain a hardcoded gas limit value', ;() => {
+        const GAS_LIIMIT = '79804' // fix if sending 0.000004 GNO to diogo.eth
+
+        // Click Advanced parameters
+        cy.contains('Estimated fee price').click()
```

Comment:
Can we also check that the gas price is loaded?

Reply:
I've added an assertion to assure the gas price is loaded